test/run/test/exec-args.test.ts

```diff
@@ -0,0 +1,29 @@
+import { expect, test } from 'vitest'
+import { execa } from 'execa'
+import { runVitest } from '../../test-utils'
+
+test.each([
+  { pool: 'forks', execArgv: ['--hash-seed=1', '--random-seed=1', '--no-opt'] },
+  { pool: 'threads', execArgv: ['--inspect-brk'] },
+  { pool: 'vmThreads', execArgv: ['--inspect-brk'] },
+] as const)('should pass execArgv to { pool: $pool } ', async ({ pool, execArgv }) => {
+  const fileToTest = `exec-args-fixtures/${pool}.test.ts`
+
+  const vitest = await runVitest({
+    include: [fileToTest],
+    pool,
+    poolOptions: {
+      [pool]: {
+        execArgv,
+      },
+    },
+  })
+
+  expect(vitest.stdout).toContain(`✓ ${fileToTest}`)
+})
+
+test('should not pass execArgv to workers when not specified in the config', async () => {
+  const { stdout } = await execa('npx', ['vitest', '--run', '--root', 'no-exec-args-fixtures', '--config', 'vitest.config.ts'])
+
+  expect(stdout).toContain('✓ no-exec-argv.test.ts')
+})
```

Comment:
Let's change this to following:

```ts
test('should not pass execArgv to workers when not specified in the config', async () => {
  const { stdout, stderr } = await execa('node', [
    '--title', 'this-works-only-on-main-thread',
    './node_modules/vitest/vitest.mjs', '--run',
    '--root', 'no-exec-args-fixtures',
  ], {
    reject: false,
    env: {
      VITE_NODE_DEPS_MODULE_DIRECTORIES: '/node_modules/,/packages/',
    },
  })

  expect(stderr).not.toContain('Error: Initiated Worker with invalid execArgv flags: --title')
  expect(stderr).not.toContain('ERR_WORKER_INVALID_EXEC_ARGV')
  expect(stdout).toContain('✓ no-exec-argv.test.ts')
})
```

When the implementation breaks, this tests shows following:

```
 FAIL  test/exec-args.test.ts > should not pass execArgv to workers when not specified in the config
AssertionError: expected '\n⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯\nError: Initiated Worker with invalid execArgv flags: --title ... Serialized Error: { code: \'ERR_WORKER_INVALID_EXEC_ARGV\' }\n' to not include 'Error: Initiated Worker with invalid execArgv flags: --title'
 ❯ test/exec-args.test.ts:37:22
     35|   })
     36| 
     37|   expect(stderr).not.toContain('Error: Initiated Worker with invalid execArgv flags: --title')
       |                      ^
     38|   expect(stderr).not.toContain('ERR_WORKER_INVALID_EXEC_ARGV')
     39|   expect(stdout).toContain('✓ no-exec-argv.test.ts')
```

Reply:
The environment variable `VITE_NODE_DEPS_MODULE_DIRECTORIES` is from `.npmrc` which is automatically picked when running scripts through `npm`. But as this test would invoke `node` directly it needs to be added here manually.

https://github.com/vitest-dev/vitest/blob/b3bc866d441eeead32ab5503d2f9565e25c0a0d1/.npmrc#L5

Reply:
That works well!
Thank you for the idea and the explanation